### PR TITLE
Rename `ILBuilder` to `FunctionBuilderContext`

### PR DESCRIPTION
**Pull request by afnanenayet**
Hey, this is my first PR with the Cretonne project, please let me know if there's something I should do that I didn't.

I picked up issue [#266](https://github.com/Cretonne/cretonne/issues/266) and renamed ILBuilder
to `FunctionBuilderContext` as per the suggestion from the issue.

I ran `test-all.sh` and everything seems to be fine.

### CODE REVIEW

File: lib/frontend/src/frontend.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ where
     /// Source location to assign to all new instructions.
     srcloc: ir::SourceLoc,
 
-    builder: &'a mut ILBuilder<Variable>,
+    builder: &'a mut FunctionBuilderContext<Variable>,
```
Comment (sunfishcode) [inline lib/frontend/src/frontend.rs]: I see you updated an `il_builder` variable below; we can also rename the `builder` field here.

File: lib/wasm/src/func_translator.rs [hunk 2]
```diff
@@ -20,15 +20,15 @@ use wasmparser::{self, BinaryReader};
 /// by a `FuncEnvironment` object. A single translator instance can be reused to translate multiple
 /// functions which will reduce heap allocation traffic.
 pub struct FuncTranslator {
-    il_builder: ILBuilder<Variable>,
+    il_builder: FunctionBuilderContext<Variable>,
```
Comment (sunfishcode) [inline lib/wasm/src/func_translator.rs]: Similarly, we should rename this `il_builder` field too.

**Comment by phase** [comment]
@afnanenayet 

You created the commits with the email `afnan@blend.com`, which you don't have registered to your GitHub account. If you go to your [Email Settings](https://github.com/settings/emails), you can add it to the list and GitHub will recognize that the commits were created by your account, if you want them to be.

**Comment by afnanenayet** [comment]
@phase I used my work laptop to push those commits earlier, I don't want to let my work email manage my personal GitHub, I have my email settings all set for my personal laptops

**Comment by sunfishcode** [comment]
Looks good!

**Pull request merged**